src/runtime_src/core/common/utils.cpp

```diff
@@ -183,24 +185,57 @@ unit_convert(size_t size)
 uint16_t
 bdf2index(const std::string& bdfstr, bool _inUserDomain)
 {
-  auto n = std::count(bdfstr.begin(), bdfstr.end(), ':');
-
-  std::stringstream s(bdfstr);
-  uint16_t dom = 0, b = 0, d = 0, f = 0;
-  char dummy;
-
-  if (n == 2)
-    s >> std::hex >> dom >> dummy;
-  s >> std::hex >> b >> dummy >> d >> dummy >> f;
-
-  if ((n != 1 && n != 2) || s.fail())
-    throw std::runtime_error("Bad BDF string '" + bdfstr + "'");
+  std::vector<std::string> tokens; 
+  boost::split(tokens, bdfstr, boost::is_any_of(":")); 
+  int radix = 16;
+  uint16_t bus = 0, dev = 0, func = std::numeric_limits<uint16_t>::max();
```

Comment:
Help me understand, what does it mean to set the "func" value to 0xffff?

I thought it is either '0' or '1' depending on whether or not the mgmt or user drivers are being talked to.   If not defined, should these values be used instead?

Reply:
This "func" value is what we get from the user. If they pass in a 0 or a 1, we parse it, but if they don;t pass in any function, then the value is -1. The "cmp_func" checks if this variable has a valid value, i.e., not -1 and parses it if that is the case.